### PR TITLE
- add ninja

### PR DESCRIPTION
**Pull request by SSE4**
Signed-off-by: SSE4 <tomskside@gmail.com>

Specify library name and version:  **ninja/1.9.0**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/ninja/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
```
Comment (Croydon) [inline recipes/ninja/all/conanfile.py]: remove this two meta lines please

File: recipes/ninja/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+class NinjaConan(ConanFile):
+    name = "ninja"
+    version = "1.9.0"
+    description = "Ninja is a small build system with a focus on speed"
+    license = "Apache-2.0"
+    url = "https://github.com/bincrafters/conan-ninja_installer"
+    homepage = "https://github.com/ninja-build/ninja"
+    author = "Bincrafters <bincrafters@gmail.com>"
+    export = ["LICENSE.md"]
```
Comment (Croydon) [inline recipes/ninja/all/conanfile.py]: remove this line

File: recipes/ninja/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+import os
+from conans import ConanFile, tools, AutoToolsBuildEnvironment
+
+
+class NinjaConan(ConanFile):
+    name = "ninja"
+    version = "1.9.0"
+    description = "Ninja is a small build system with a focus on speed"
+    license = "Apache-2.0"
+    url = "https://github.com/bincrafters/conan-ninja_installer"
```
Comment (Croydon) [inline recipes/ninja/all/conanfile.py]:
```suggestion
    url = "https://github.com/conan-io/conan-center-index"
```

File: recipes/ninja/all/test_package/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
```
Comment (Croydon) [inline recipes/ninja/all/test_package/conanfile.py]: remove this two meta lines please

File: recipes/ninja/all/conanfile.py (hunk 2 above)
Comment (uilianries) [inline recipes/ninja/all/conanfile.py]:
```suggestion
```

File: recipes/ninja/all/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,61 @@
+import os
+from conans import ConanFile, tools, AutoToolsBuildEnvironment
+
+
+class NinjaConan(ConanFile):
+    name = "ninja"
+    version = "1.9.0"
```
Comment (uilianries) [inline recipes/ninja/all/conanfile.py]:
```suggestion
```

File: recipes/ninja/all/conanfile.py [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/ninja-build/ninja"
+    author = "Bincrafters <bincrafters@gmail.com>"
+    settings = "os_build", "arch_build", "compiler"
+    _source_subfolder = "source_subfolder"
+
+    def _build_vs(self):
+        with tools.chdir(self._source_subfolder):
+            with tools.vcvars(self.settings, filter_known_paths=False):
+                self.run("python configure.py --bootstrap")
```
Comment (uilianries) [inline recipes/ninja/all/conanfile.py]:
```suggestion
                self.run("%s configure.py --bootstrap" % sys.executable)
```

File: recipes/ninja/all/conanfile.py [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+            if self.settings.os_build == "Linux":
+                if self.settings.arch_build == "x86":
+                    cxx += " -m32"
+                elif self.settings.arch_build == "x86_64":
+                    cxx += " -m64"
+            env_build = AutoToolsBuildEnvironment(self)
+            env_build_vars = env_build.vars
+            env_build_vars.update({'CXX': cxx})
+            with tools.environment_append(env_build_vars):
+                self.run("python configure.py --bootstrap")
```
Comment (uilianries) [inline recipes/ninja/all/conanfile.py]:
```suggestion
                self.run("%s configure.py --bootstrap" % sys.executable)
```

**Comment by Croydon** [comment]
For some reason installers build by CCI seems to use the new CXX11 ABI which then doesn't run in GCC 4.9 containers. Am I missing something here or is this effectively a bug? 🤔 

https://travis-ci.com/bincrafters/conan-sdl2/jobs/247592499#L1801

**Comment by ericLemanissier** [comment]
I just saw similar situation here https://travis-ci.com/bincrafters/conan-glib/jobs/247779239#L1154

**Comment by SSE4** [comment]
let's create a new issue to discuss the situation

**Pull request merged**